app/src/main/java/org/fedorahosted/freeu2f/u2f/APDUReply.java

```diff
@@ -0,0 +1,49 @@
+package org.fedorahosted.freeu2f.u2f;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class APDUReply implements Frameable {
+    public enum StatusCode implements Frameable {
+        NO_ERROR(0x9000),
+        CONDITIONS_NOT_SATISFIED(0x6985),
+        WRONG_DATA(0x6a80),
+        WRONG_LENGTH(0x6700),
+        CLA_NOT_SUPPORTED(0x6e00),
+        INS_NOT_SUPPORTED(0x6d00),
+        UNKNOWN(0x6f00);
+
+        private final char value;
+
+        StatusCode(int value) {
+            this.value = (char) value;
+        }
+
+        @Override
+        public byte[][] toFrames(int mtu) {
+            return new APDUReply(this).toFrames(mtu);
+        }
+    }
+
+    private final StatusCode code;
+    private final byte[] data;
+
+    public APDUReply(StatusCode code) {
```

Comment:
These 2 ctors expose the internal malleable reference (arrays in Java are one of the few pass by refs), could be undesirable, if something else is expecting that to remain constant.